### ISSUE TITLE
Support login with OIDC

### ISSUE THREAD

**Issue by michaelhthomas** [Fallenbagel/jellyseerr#183]
### Description

Authentication providers like Authelia, Auth0, Authentik, Keycloak, etc. make managing user permissions across selfhosted services much easier. 

### Desired Behavior

I would like to be able to configure a custom OIDC provider for logins, and have this option appear on the Jellyseer login page. [This PR](https://github.com/sct/overseerr/pull/1941) from Overseerr implements support for generic OIDC login. Since OIDC is already supported for Plex SSO, supporting other external authentication services is fairly straightforward.

### Additional Context

Here is a screencast from the linked Overseerr PR showing what this feature would look like when implemented.

https://cln.sh/wGbPIdYUKas2QId0cqxw

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

**Comment by michaelhthomas** [comment]
I'd be happy to try my hand at re-implementing this for Jellyseerr, so long as all are onboard.

**Comment by Fallenbagel** [comment]
> I'd be happy to try my hand at re-implementing this for Jellyseerr, so long as all are onboard.

Any help would be much appreciated c:

**Comment by michaelhthomas** [comment]
@Fallenbagel Would you say it's alright to go ahead and remove the Plex OAuth login stuff in here as well? It's intertwined with the authentication logic a little more than I'd like.

**Comment by Fallenbagel** [comment]
> @Fallenbagel Would you say it's alright to go ahead and remove the Plex OAuth login stuff in here as well? It's intertwined with the authentication logic a little more than I'd like.

If the plex OAuth is removed, wouldn't that remove support for plex?

**Comment by michaelhthomas** [comment]
Login with Plex, yes. If the goal is to keep that support in place, then I can definitely leave it.

**Comment by Fallenbagel** [comment]
> Login with Plex, yes. If the goal is to keep that support in place, then I can definitely leave it. I could also see if it's possible to just detangle the Plex login a bit so that it's isolated like the rest of the auth providers.

Yeah the goal is for plex to be supported as well

**Comment by michaelhthomas** [comment]
Got it. I'll try separating plex login into its own component so all the login logic isn't on the main `Login` component

**Comment by ados8** [comment]
Testing with Authentik and it fails the OIDC login due to exceeded threshold.
![image](https://user-images.githubusercontent.com/49982433/194679826-06aa8dcc-e6b0-4ede-b318-9b10e094717d.png)

Jellyseerr presents an error to sign in.
![image](https://user-images.githubusercontent.com/49982433/194679805-030efd44-6d95-4ed6-9187-99223b35d0d1.png)

My config is as follows:
OIDC Issuer URL: `https://auth.domain.com/application/o/jellyseerr/`
OIDC Provider Name: `Authntik`
OIDC Client ID: `generated ID from Authntik`

**Comment by kurokay** [comment]
It would be an awesome feature to add !

**Comment by rmccullagh99** [comment]
Definitely +1 to the interest here! I use Authentik and would love to more easily and granularly secure my *rr stack

**Comment by Fallenbagel** [comment]
> Definitely +1 to the interest here! I use Authentik and would love to more easily and granularly secure my *rr stack

Could you test out the `:preview-184` tag. Make sure to not test on your production system

**Comment by rmccullagh99** [comment]
Thanks @Fallenbagel! I tested a setup with Authentik and it's worked great!
Only thing to note is the oidc-authenticated user is shown as type 'local', and a the email (rather than username) is used as displayname (not sure if this is easy to change - I don't really mind all that much)
![jellyseerr_oidc](https://user-images.githubusercontent.com/42370113/205236293-118c5839-111e-4c32-9632-1b09e031ebd1.jpg)

**Comment by Sapd** [comment]
So I have :preview-184 running with only SSO login since mid of December without any problems. (Also with Authentik)

**Comment by fistwho** [comment]
Thank you. I tested it with Authentik it works like a charm. I hope we can get it in a stable release soon.

**Comment by muppie** [comment]
What return_uri did you use? I cannot get it to work

**Comment by fistwho** [comment]
> What return_uri did you use? I cannot get it to work

In authentik i use: https://jellyseerr.yourdomain.com/login/oidc/callback

**Comment by Fallenbagel** [comment]
> Hey everyone! I'm not super familiar with the tags on Github. Will this be available in the normal release branch or is it still just available in :preview-184?

It's not available in `:preview-OIDC` as indicated in my previous comment

**Comment by thimplicity** [comment]
> > What return_uri did you use? I cannot get it to work
> 
> In authentik i use: https://jellyseerr.yourdomain.com/login/oidc/callback

Whan using this (and every other one so far) I get "The request fails due to a missing, invalid, or mismatching redirection URI (redirect_uri)."

Any ideas?

**Comment by M-Davies** [comment]
> > > What return_uri did you use? I cannot get it to work
> > 
> > 
> > In authentik i use: https://jellyseerr.yourdomain.com/login/oidc/callback
> 
> When using this (and every other one so far) I get "The request fails due to a missing, invalid, or mismatching redirection URI (redirect_uri)."
> 
> Any ideas?

Had a similar issue with the Jellyfin SSO plugin https://github.com/9p4/jellyfin-plugin-sso/discussions/154

Apparantly disabling endpoint verification can help on the Jellyfin side but it's probably not recommended that it's enabled in actual production environments

**Comment by bobobado** [comment]
First off, thank you so much for your work on this. It is extremely appreciated as I transition from Plex and Overseer to Jellyfin and Jellyseer! Would it be possible to implement a redirect to the authentik page? For example with Audiobookshelf, you can set "Auto Launch" so if a user accesses the Audiobookshelf URL, it redirects to the OIDC provider's page instead. Once authenticated, the user is then redirected to Audiobookshelf. Happy to help test if this is doable!

**Comment by Fallenbagel** [comment]
> The two login pages is what I'm trying to avoid. By using the "Auto Launch" feature in Audiobookshelf, users authenticate once and are immediately taken into the service. This is what I was hoping for

The integrated solution is better though. Why have it redirect to authentik page to authenticate when it can just do a callback and authenticate with authentik in jellyseerr itself?
I still dont understand why thats not a better option for you.

**Comment by bobobado** [comment]
I totally get what you're saying and maybe I'm misunderstanding how to accomplish this with the tools currently available but my goal is a single authentication page for several services. By going to the Authentik embedded outpost, a user can authenticate and immediately launch the Audiobookshelf service, without having to authenticate again. The way I understand it now, if a user were to login to the embedded outpost and then select the jellyseer application, they will have to login again or at the very least, click the OIDC button at the jellyseer page. Please enlighten me if I'm missing something. This might just be personal preference

**Comment by Fallenbagel** [comment]
Doesnt the current OIDC work like you click the oidc button and it logs you in

**Comment by bobobado** [comment]
Yes, if the user has authenticated with Authentik in another window previously

**Comment by Fallenbagel** [comment]
So whats the issue then 🤔

**Comment by bobobado** [comment]
I wouldn't call it an issue. If the user authenticates with Authentik and then goes to jellyseer and clicks the OIDC button, it logs them in to jellyseer. One additional click isn't a huge deal, I just liked the simplicity of launching the app from the embedded outpost and immediately being taken to the application. Again, this might just be a personal preference I have. I appreciate your work on this greatly and your willingness to discuss this.

**Comment by Fallenbagel** [comment]
> I wouldn't call it an issue. If the user authenticates with Authentik and then goes to jellyseer and clicks the OIDC button, it logs them in to jellyseer. One additional click isn't a huge deal, I just liked the simplicity of launching the app from the embedded outpost and immediately being taken to the application. Again, this might just be a personal preference I have. I appreciate your work on this greatly and your willingness to discuss.

What happens if the user didnt authenticate then pressed oidc button. Does that open an authentik window to authenticate?

**Comment by bobobado** [comment]
Yes. That works perfectly

**Comment by Fallenbagel** [comment]
> Yes. That works perfectly

Thanks for clarifying. I haven't tested or reviewed that pr yet 😅

**Comment by bobobado** [comment]
Once I knew what to do in Authentik, it was a breeze. Thank you so much again for implementing this in the first place!

**Comment by Fallenbagel** [comment]
> Once I knew what to do in Authentik, it was a breeze. Thank you so much again for implementing this in the first place!

Thanks goes to @michaelhthomas c:

**Comment by SadRobot11** [comment]
Can someone kindly explain to me, how can I access this feature? Is this not part of the :latest release yet?

**Comment by Fallenbagel** [comment]
> Can someone kindly explain to me, how can I access this feature? Is this not part of the :latest release yet?

You need to use the tag `:preview-OIDC`

**Comment by SadRobot11** [comment]
I am trying to set this up with Authelia, and I get this error
```
{
"error":"invalid_request",
"error_description":"The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls."}
```
What kind of redirect uris should I set up in Authelia config for jellyseerr?

**Comment by Fallenbagel** [comment]
> I am trying to set this up with Authelia, and I get this error
> 
> > The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls.
> 
> What kind of redirect uris should I set up in Authelia config for jellyseerr?

You should ask this in #184

**Comment by BartvanLaar** [comment]
This does not seem to work on the RPI 5 as i get an Exec format error when using the :preview-OIDC tag

**Comment by Fallenbagel** [comment]
> This does not seem to work on the RPI 5 as i get an Exec format error when using the :preview-OIDC tag.
> 
> I would've loved to try and test this though.

Because the preview tag doesn't build for arm. You could build a preview tag on the pi itself or cross build too

**Comment by BartvanLaar** [comment]
> > This does not seem to work on the RPI 5 as i get an Exec format error when using the :preview-OIDC tag.
> > I would've loved to try and test this though.
> 
> Because the preview tag doesn't build for arm. You could build a preview tag on the pi itself or cross build too
> 
> Or even run natively after checking out to this pr

When I try to do this, on both windows and the Raspberry Pi 5, I get the following error (both on dev and pr branches);
![image](https://github.com/Fallenbagel/jellyseerr/assets/52526138/41afe520-12fc-4887-91fd-6bf4e8d0b906)

Is this something i should follow readme for or should this have worked from the get go?

**Comment by brenninger** [comment]
I am running into an issue signing into Jellyseerr using Authentik. After I login on Jellyseerr using my Authentik OIDC login, it is then stuck in a spinning wheel of death saying "Authentication in progress." Has anyone had this issue and know what I am missing. I'm not seeing anything pop up in the logs.

**Comment by marouamghar** [comment]
FYI, it's failing to auth with the latest versions of Authelia (master branch, or the latest tag for 4.38 beta), because of these two errors: "request.cookies should have required property 'oidc-state'" "cookie 'connect.sid' required"

**Comment by j007bond007** [comment]
When will this be part of the stable/main release? This is the one feature holding me back from migrating from Ombi. Thankyou!

**Comment by keesfluitman** [comment]
> When will this be part of the stable/main release? This is the one feature holding me back from migrating from Ombi. Thankyou!

i cant get ombi header auth working either with authentik...how did you get that to work? It just refuses to authenticate using a header. Thanks for this preview. Ill test it.

**Comment by nebb00** [comment]
I also very much want this

**Comment by Marc-Proux** [comment]
Same here. All my apps are managed by Authentik so I'm very interested

**Comment by keesfluitman** [comment]
> Same here. All my apps are managed by Authentik so I'm very interested

also still running jellyseerr on this preview version.

**Comment by HumnResources** [comment]
> > Same here. All my apps are managed by Authentik so I'm very interested
> 
> also still running jellyseerr on this preview version.

Are you able to expand o  how you managed to make it work? What version of Authentik are you running? I got the redirect working however it just shows a red error message. Logs just say OIDC unknown error ... code 405. Same as noted by another user above. 

Jellyseerr is far better than Ombi IMO, however I'm struggling to get this to work. This is definitely all it needs from me to switch.

**Comment by keesfluitman** [comment]
> > > Same here. All my apps are managed by Authentik so I'm very interested
> > 
> > 
> > also still running jellyseerr on this preview version.
> 
> Are you able to expand o how you managed to make it work? What version of Authentik are you running? I got the redirect working however it just shows a red error message. Logs just say OIDC unknown error ... code 405. Same as noted by another user above.
> 
> Jellyseerr is far better than Ombi IMO, however I'm struggling to get this to work. This is definitely all it needs from me to switch.

Send me a pm on discord. Same name.

**Comment by HumnResources** [comment]
> > > > Same here. All my apps are managed by Authentik so I'm very interested
> > > 
> > > 
> > > also still running jellyseerr on this preview version.
> > 
> > 
> > Are you able to expand o how you managed to make it work? What version of Authentik are you running? I got the redirect working however it just shows a red error message. Logs just say OIDC unknown error ... code 405. Same as noted by another user above.
> > Jellyseerr is far better than Ombi IMO, however I'm struggling to get this to work. This is definitely all it needs from me to switch.
> 
> Send me a pm on discord. Same name.

Unfortunately can't send a message without sharing a server, searching doesn't come up with anything.

**Comment by keesfluitman** [comment]
> > > > > Same here. All my apps are managed by Authentik so I'm very interested
> > > > 
> > > > 
> > > > also still running jellyseerr on this preview version.
> > > 
> > > 
> > > Are you able to expand o how you managed to make it work? What version of Authentik are you running? I got the redirect working however it just shows a red error message. Logs just say OIDC unknown error ... code 405. Same as noted by another user above.
> > > Jellyseerr is far better than Ombi IMO, however I'm struggling to get this to work. This is definitely all it needs from me to switch.
> > 
> > 
> > Send me a pm on discord. Same name.
> 
> Unfortunately can't send a message without sharing a server, searching doesn't come up with anything.
sorry missed this. we should pm somehow. mail me, my fullname at protonmail.com

**Comment by vutsalsinghal** [comment]
guys, which branch can I use to test this? `preview-OIDC` is not there anymore. I want to switch to OIDC on jellyseerr so I can disable user/pass on both jellyfin/seerr. Appreciate any info 🍻

**Comment by JoshuaSeidel** [comment]
> guys, which branch can I use to test this? `preview-OIDC` is not there anymore. I want to switch to OIDC on jellyseerr so I can disable user/pass on both jellyfin/seerr. Appreciate any info 🍻

its a tag not a branch: https://github.com/fallenbagel/jellyseerr/tree/preview-OIDC

**Comment by ZSMB1** [comment]
Oh, thanks. Could have sworn I checked it a few days ago and have not found such a tag. Maybe it was on docker hub.
 Btw, it would be nice to have a container published with the same tag.

**Comment by fallenbagel** [comment]
> Oh, thanks. Could have sworn I checked it a few days ago and have not found such a tag. Maybe it was on docker hub.
>  Btw, it would be nice to have a container published with the same tag.

There is?

**Comment by vutsalsinghal** [comment]
> > guys, which branch can I use to test this? `preview-OIDC` is not there anymore. I want to switch to OIDC on jellyseerr so I can disable user/pass on both jellyfin/seerr. Appreciate any info 🍻
> 
> its a tag not a branch: https://github.com/fallenbagel/jellyseerr/tree/preview-OIDC

Thanks mate. I have OIDC up and running (using authentik). Hope it'll be merged with master soon 🖖🏼

**Comment by enieuwy** [comment]
This is very well-executed, and I've decided to swap from Overseerr for this very feature. Looking foward to the merge.

**Comment by j007bond007** [comment]
So excited for this to be re-added now that the linked account page is done :)

**Comment by animalillo** [comment]
i'm triying to use this branch but i get those errors when attempting to use it:

[error][Jellyfin Sync]: Sync interrupted {"errorMessage":"SQLITE_ERROR: no such column: User_settings.region"}

Any ideas?

**Comment by fallenbagel** [comment]
> i'm triying to use this branch but i get those errors when attempting to use it:
> 
> [error][Jellyfin Sync]: Sync interrupted {"errorMessage":"SQLITE_ERROR: no such column: User_settings.region"}
> 
> Any ideas?

That branch should not be used. It's deprecated and outdated. If you want to use you will need to use a fresh config. It's based on jellyseerr v1

**Comment by enieuwy** [comment]
With my limited git knowledge, I thought the OIDC work had been incorporated to the develop tag, so I switched. Looks like I was wrong? Trying to decide whether to switch back to the outdated version. (I'll have to nuke my config to avoid the above error.)

**Comment by vutsalsinghal** [comment]
> With my limited git knowledge, I thought the OIDC work had been incorporated to the develop tag, so I switched. Looks like I was wrong? Trying to decide whether to switch back to the outdated version. (I'll have to nuke my config to avoid the above error.)

you need to use the following image tag to try OIDC - `fallenbagel/jellyseerr:preview-OIDC`. Its recommended that you do setup a fresh env.

**Comment by enieuwy** [comment]
I had it running on that tag, but fallenbagel noted that it was very outdated. Looks like I switched to develop too early.

**Comment by DuvelCorp** [comment]
> > With my limited git knowledge, I thought the OIDC work had been incorporated to the develop tag, so I switched. Looks like I was wrong? Trying to decide whether to switch back to the outdated version. (I'll have to nuke my config to avoid the above error.)
> 
> you need to use the following docker image tag to try OIDC - `fallenbagel/jellyseerr:preview-OIDC`. Its recommended that you do setup a fresh env. I have it up and running for ~1mo now with Authentik. No issues at all so far!

Quick question as its not really clear to me :
When switching to Authentik for the login, does it also affect new user's creation in Jellyseer, or is its still relying on copying users from Jellyfin ?

**Comment by vutsalsinghal** [comment]
> > > With my limited git knowledge, I thought the OIDC work had been incorporated to the develop tag, so I switched. Looks like I was wrong? Trying to decide whether to switch back to the outdated version. (I'll have to nuke my config to avoid the above error.)
> > 
> > 
> > you need to use the following docker image tag to try OIDC - `fallenbagel/jellyseerr:preview-OIDC`. Its recommended that you do setup a fresh env. I have it up and running for ~1mo now with Authentik. No issues at all so far!
> 
> Quick question as its not really clear to me : When switching to Authentik for the login, can it also trigger the creation of new users in Jellyseer if they dont exists yet ? Or it can't create users and is still relying on copying users from Jellyfin, and thus the SSO login would be based on usernames that should be the same in Authentik and Jellyfin/Jellyseer ?

You dont need to create/import users (for fallenbagel/jellyseerr:preview-OIDC image). Go to user settings > users > Enable OIDC Sign-In 

Click on the gear icon next to "Enable OIDC Sign-In" and enable "Allow Jellyfin Usernames" under Advanced. It'll seamlessly let users login to connect to jellyseer without having to create/import them!

**Comment by DuvelCorp** [comment]
ok great. 
I am going to wait for this to be merged into the master branch before using it because I have a dozen of users and I understand from above posts that the preview-OIDC tag doesn't use the latest database version and I dont want a fresh install to keep all my users data as it is now

**Comment by nebb00** [comment]
With the latest Plex news, really hoping for this OIDC support so I can migrate my users

**Comment by nebb00** [comment]
Yeah sorry I should have attached. Worst case will just have to use local accounts... meh

**Comment by kassarole** [comment]
I'm testing the linked PR and getting an error when I try to sign in
`{
 "cause": "Unknown error",
 "errorMessage": "SQLITE_ERROR: no such table: linked_accounts",
 "ip": "10.0.25.1"
}`
I've tried recreating the container to rebuild the DB but still having the same issue.

**Comment by fallenbagel** [comment]
> I'm testing the linked PR and getting an error when I try to sign in
> `{
>  "cause": "Unknown error",
>  "errorMessage": "SQLITE_ERROR: no such table: linked_accounts",
>  "ip": "10.0.25.1"
> }`
> I've tried recreating the container to rebuild the DB but still having the same issue.

With a fresh config? Recreating container wouldn't clear config (you dont have to recreate container. You can just stop and delete config and restart)

**Comment by kassarole** [comment]
> > I'm testing the linked PR and getting an error when I try to sign in
> > `{ "cause": "Unknown error", "errorMessage": "SQLITE_ERROR: no such table: linked_accounts", "ip": "10.0.25.1" }`
> > I've tried recreating the container to rebuild the DB but still having the same issue.
> 
> With a fresh config? Recreating container wouldn't clear config (you dont have to recreate container. You can just stop and delete config and restart)
Yup fully deleted the config and recreated

**Comment by kassarole** [comment]
Checking my database I can confirm that table is definitely not there even after deleting the config folder and letting the container remake it

**Comment by michaelhthomas** [comment]
With a brand new database (like after deleting the entire config folder), this should not be an issue. However, I haven't added database migrations to the branch yet, so if you're coming from an existing database, it will not add the new table. If you'd like to add it manually, you can do something like

```sql
CREATE TABLE "linked_accounts" ("id" integer PRIMARY KEY AUTOINCREMENT NOT NULL, "provider" varchar(255) NOT NULL, "sub" varchar(255) NOT NULL, "username" varchar NOT NULL, "userId" integer, CONSTRAINT "FK_2c77d2a0c06eeab6e62dc35af64" FOREIGN KEY ("userId") REFERENCES "user" ("id") ON DELETE CASCADE ON UPDATE NO ACTION)
```

but I will be adding the migrations shortly, so it may be better just to hold off until then.

**Comment by kassarole** [comment]
Gotcha I'm not sure what the issue was but manually creating the table with the SQL you sent fixed it for me. Thanks!

**Comment by v3DJG6GL** [comment]
Has someone tried to migrate directly from #184 to #1505 ?
I've been running this super outdated Jellyseerr version the last 1.5 years and I'd like to try out the latest OIDC PR. But if possible, not loosing all requests and userdata... 🐵

**Comment by cordlord** [comment]
@v3DJG6GL My recommendation would be to make a copy of the data directory and start up separate test container using the copied directory to see what happens. You'll probably have to wait a bit for them to push the migration fix to add the extra table, but then you should be good to test.

I'm sure there will be others doing this, but there's only one way to find out if you don't want to wait.

**Comment by smdion** [comment]
> With a brand new database (like after deleting the entire config folder), this should not be an issue. However, I haven't added database migrations to the branch yet, so if you're coming from an existing database, it will not add the new table. If you'd like to add it manually, you can do something like
> 
> CREATE TABLE "linked_accounts" (
>   "id" integer PRIMARY KEY AUTOINCREMENT NOT NULL,
>   "provider" varchar(255) NOT NULL,
>   "sub" varchar(255) NOT NULL,
>   "username" varchar NOT NULL,
>   "userId" integer,
>   CONSTRAINT "FK_2c77d2a0c06eeab6e62dc35af64" FOREIGN KEY ("userId") REFERENCES "user" ("id") ON DELETE CASCADE ON UPDATE NO ACTION
> )
> but I will be adding the migrations shortly, so it may be better just to hold off until then.

I can also confirm this works for an existing db.

**Comment by vutsalsinghal** [comment]
> 
> I can also confirm this works for an existing db.

It worked for me as well (manually added the SQL above) but getting "Redirect URI Error" with Authentik. Are you using authentik? whats your redirect URI? Its "http://domain.com/login/oidc/callback" for me. Was working with this redirect uri before the update :(

**Comment by McGeaverBeaver** [comment]
@vutsalsinghal  , This is not required. 

Authentik works just fine, You are indeed using the wrong callback ID...double check it.. What I found is after I did the above table add, I still had issues...

1) I modified my settings.json to  "   "mediaServerLogin": true, " 
2) Logged in with plex
3) Found that OIDC config was not setup in the GUI yet was still in my json file. 
4) Re configured ODIC from the GUI , I set my SLUG manually, YET it still changed it 
5) updated my callback URL for the new slug it set.. 
6) closed all windows, opened up and it worked again as expected.

**Comment by smdion** [comment]
> > [@vutsalsinghal](https://github.com/vutsalsinghal) , This is not required.
> > Authentik works just fine, You are indeed using the wrong callback ID...double check it.. What I found is after I did the above table add, I still had issues...
> > 
> > 1. I modified my settings.json to  "   "mediaServerLogin": true, "
> > 2. Logged in with plex
> > 3. Found that OIDC config was not setup in the GUI yet was still in my json file.
> > 4. Re configured ODIC from the GUI , I set my SLUG manually, YET it still changed it
> > 5. updated my callback URL for the new slug it set..
> > 6. closed all windows, opened up and it worked again as expected.
> 
> I did setup OIDC from jellyseer UI again but didn't update the slug in authentik. You're right, it changed the slug even though I set it to something else! I'll give it another try by updating the slug in authentik. But is the redirect url endpoint correct `http://domain.com/login/oidc/callback` ?

The redirect URI changed as well. In authentik you can just remove what you have there and it will add the correct one.  If not, this is what mine is

https://%jellyseerrsubdomain%.%domain%.com/login/oidc/callback/%providername%

**Comment by dominicrico** [comment]
Found another bug while testing this behind a revers proxy. The redirect url is not set correct if using this with ssl/https. The redirect url is build with http not https even though I set an application url from the general tab. It should even use the application url from the settings or the one im currently on to create the provider.

**Comment by vutsalsinghal** [comment]
> Found another bug while testing this behind a revers proxy. The redirect url is not set correct if using this with ssl/https. The redirect url is build with http not https even though I set an application url from the general tab. It should even use the application url from the settings or the one im currently on to create the provider.

That's not true for me. I'm using Authentik + nginx with letsEncrypt certs setup. As @smdion mentioned on 03/25/25, the redirect url is automatically set to something like `https://domain.com/login/oidc/callback/custom-slug`.

**Comment by dominicrico** [comment]
> > Found another bug while testing this behind a revers proxy. The redirect url is not set correct if using this with ssl/https. The redirect url is build with http not https even though I set an application url from the general tab. It should even use the application url from the settings or the one im currently on to create the provider.
> 
> That's not true for me. I'm using Authentik + nginx with letsEncrypt certs setup. As [@smdion](https://github.com/smdion) mentioned on 03/25/25, the redirect url is automatically set to something like `https://domain.com/login/oidc/callback/custom-slug`.

Weird, I have exactly the same setup but the redirect uri is build with http ...

**Comment by gauthier-th** [comment]
Please use the [dedicated discussion](https://github.com/fallenbagel/jellyseerr/discussions/1529)
